### DISCUSSION TITLE
Bounded useStore hook for vanilla stores - not working as expected

### DISCUSSION THREAD

**Discussion by marcolink** [pmndrs/zustand#1735] [General] [answered]
Hello 👋 

first of all - thanks for my favorite state solution 🙏 

I'm currently migrating our project to use v4. 

All our stores are defined as vanilla stores (this helps for different test scenarios), and additional are exported as react hook stores using `import create from 'zustand';` Ass I understand, this is still working, but not recommended anymore in v4. 

When following the instruction [here](https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md#bounded-usestore-hook-for-vanilla-stores), the resulting hooks for the stores are not correctly typed. In fact, the state is `never` 🤔 

I [copied your example code into the typescript playground](https://www.typescriptlang.org/play?ts=5.0.3#code/JYWwDg9gTgLgBAbzgVwM4FMDKNroDRza4CCYwcAvnAGZQQhwDkAXmjAIYB2AJowFChIsRHADGUdOxhYcEyjToMWbLtwD0ANy7AANjvb8BnaVGrtR6OACFJUbFMsI+cOACNbqAFxxOyEO6hnOGBOcUkMbwAKVwBPb19-dCgASjgAXgA+OA0IYG4+Cj4+UQhOVHh3djtZSzSxCQciCQAeGyr7aQzI5MjIjBhUzLhIpxdKqC84AAY8IJCw9gjh2MGs-t7yh1XhpHHJzekAOj24AGo3GMpk5NmKa6KSsvgF6SsIZB50bgBVDCba4Z9GrbProHToUSyAjoACOyHYOlQqyCaBkuCBuAIGHBkOgcEW+M4MWhcIRSNSBOamDg6AAHtIeKhCDVSMBmh8ANacCAAd04GS6QXKuG8mD421Gw2S3gAovSoOYYB10FSMkFmgAVQUuFzYiGyAD8UQO6Fl8sVytV2w1sx1NNJiKNw3Y3htbld21cEAg4K4QWlcA1BSKMBiYEscpgCshlswWTq1LpDO4TKQAHN0DAotsQtQknAABryA2FuDxdAaJIPUrlFAYNrVXDpeqSV7vT4-P41aK2f7JatPOAAYXokE46GMze66SyTiCj1rJzqqIb-yBDhncBNxw8-aCEhgyCgnDcHkOOGwUBCae6RSoQA) (1:1) - and it also shows the problem. 

I'm currently on `typescript@5.0.3` and `zustand@4.3.7`

**Comment by JacobWeisenburger** [comment, marked as answer]
I made a PR for this: https://github.com/pmndrs/zustand/pull/1737

**Reply by marcolink** [reply]
@JacobWeisenburger thank you 🙏  - solved!
